cmd/geth/snapshot.go

```diff
@@ -576,3 +626,356 @@ func dumpState(ctx *cli.Context) error {
+	stack, _ := makeConfigNode(ctx)
+	defer stack.Close()
+
+	chaindb := utils.MakeChainDatabase(ctx, stack, true)
+	headBlock := rawdb.ReadHeadBlock(chaindb)
+	if headBlock == nil {
+		log.Error("Failed to load head block")
+		return errors.New("no head block")
+	}
+	if ctx.NArg() > 1 {
+		log.Error("Too many arguments given")
+		return errors.New("too many arguments")
+	}
+	var (
+		root common.Hash
+		err  error
+	)
+	if ctx.NArg() == 1 {
+		root, err = parseRoot(ctx.Args()[0])
+		if err != nil {
+			log.Error("Failed to resolve state root", "error", err)
+			return err
+		}
+		log.Info("Start traversing the state", "root", root)
+	} else {
+		root = headBlock.Root()
+		log.Info("Start traversing the state", "root", root, "number", headBlock.NumberU64())
+	}
+
+	var (
+		accounts   int
+		lastReport time.Time
+		start      = time.Now()
+		rangeStart = ctx.Uint64(utils.VerkleConversionInsertRangeStartFlag.Name)
+		rangeEnd   = rangeStart + ctx.Uint64(utils.VerkleConversionInsertRangeSizeFlag.Name)
+	)
+
+	if rangeEnd > 256 {
+		rangeEnd = 256
+	}
+
+	convdb, err := rawdb.NewLevelDBDatabase("verkle", 128, 128, "", false)
+	if err != nil {
+		panic(err)
+	}
+
+	flushCh := make(chan verkle.VerkleNode)
+	saveverkle := func(node verkle.VerkleNode) {
+		flushCh <- node
+	}
+	go func() {
+		for node := range flushCh {
+			comm := node.ComputeCommitment()
+			s, err := node.Serialize()
+			if err != nil {
+				panic(err)
+			}
+			commB := comm.Bytes()
+			if err := convdb.Put(commB[:], s); err != nil {
+				panic(err)
+			}
+		}
+	}()
+
+	snaptree, err := snapshot.New(chaindb, trie.NewDatabase(chaindb), 256, root, false, false, false)
+	if err != nil {
+		return err
+	}
+	accIt, err := snaptree.AccountIterator(root, common.Hash{})
+	if err != nil {
+		return err
+	}
+	defer accIt.Release()
+
+	type treeHugger struct {
+		node *verkle.LeafNode
+		stem []byte
+	}
+	treeHuggers := make([]chan *treeHugger, runtime.NumCPU())
+	subRoots := make([]*verkle.InternalNode, runtime.NumCPU())
+	rootPerCPU := 256 / runtime.NumCPU()
+	for i := range treeHuggers {
+		treeHuggers[i] = make(chan *treeHugger, 128)
+		subRoots[i] = verkle.New().(*verkle.InternalNode)
+
+		// save references for the goroutine to capture
+		hugger := treeHuggers[i]
+		root := subRoots[i]
+
+		go func() {
+			ticker := time.NewTicker(10 * time.Second)
+			defer ticker.Stop()
+			for hug := range hugger {
+				select {
+				case <-ticker.C:
+					// Check the memory usage every 10 seconds. If it
+					// goes above a given watermark, flush the tree's
+					// lower nodes to disk.
+					v, _ := mem.VirtualMemory()
+
+					// Compute flushing depth
+					// 1 if > 80%, 2 if 80% < && > 60%...
+					// don't bother cleaning up above 60%
+					depth := 5 - uint8(v.UsedPercent/20)
+
+					if depth < 3 {
+						root.FlushAtDepth(depth, saveverkle)
+					}
+				default:
+				}
+
+				if uint64(hug.stem[0]) < rangeStart && uint64(hug.stem[0]) > rangeEnd {
+					// skip stem outside the insertion range
```

Comment:
Isn't this a programming error worthy of `panic`?

Reply:
this is meant to compute sections of the tree in parallel on several machines, so one machine will only care about keys in a specific range and ignore the other values.